Clarify `data_returned`, `data_available`, and `more_data_available` (#220)
Materials-Consortia/OPTiMaDe: optimade.rst

  - **api\_version**: a string containing the version of the API implementation.
  - **time\_stamp**: a timestamp containing the date and time at which the query was executed.
  - **data\_returned**: an integer containing the total number of data resource objects returned for the current :query-param:`filter` query, independent of pagination.

>   - **more\_data\_available**: :field-val:`false` if all data for this :query-param:`filter` query has been returned in the response, and :field-val:`true` if not.

Reviewer: I would like to stress that `more_data_available : False` does not necessary mean that the current response contains all the data matching the `filter` query. To be precise, `more_data_available : False` means either that:

1. All entries matching the `filter` query is returned in the `data` payload of the current response;
2. We have reached the last page of the paginated result.

This was the original intent of `more_data_available`. It would be great to explain this somehow concisely here.

Author: Ah, agree. I missed this definition, thanks :+1:

>   - **more\_data\_available**: :field-val:`false` if all data resource objects for this :query-param:`filter` query has been returned in the response or it is the last page if pagination is implemented, and :field-val:`true` if not.

Reviewer 2: Some pickiness, think this is marginally clearer

```suggestion
  - **more\_data\_available**: :field-val:`false` if all data resource objects for this :query-param:`filter` query have been returned in the response or if it is the last page of a paginated response, and :field-val:`true` otherwise. 
```
